src/Filters/FederalState.jsx

```diff
@@ -0,0 +1,131 @@
+// It uses this name rather than 'State' to distinguish from the React state
+// Idea https://en.wikipedia.org/wiki/U.S._state
+
+import React from 'react'
+import { connect } from 'react-redux'
+import AggregationItem from './AggregationItem'
+import CollapsibleFilter from './CollapsibleFilter'
+import Typeahead from '../Typeahead'
+import { addMultipleFilters } from '../actions/filter'
+import { THESE_UNITED_STATES } from '../constants'
+import { normalize } from './utils'
+
+const buildLabel = x => {
+  return THESE_UNITED_STATES[x] + ' (' + x + ')'
+}
+
+export class FederalState extends React.Component {
+  constructor(props) {
+    super(props)
+    this._onInputChange = this._onInputChange.bind(this)
+    this._onOptionSelected = this._onOptionSelected.bind(this)
+  }
+
+  render() {
+    return (
+      <CollapsibleFilter title='State'
+                         desc='The state of the mailing address provided by the consumer'
+                         showChildren={this.props.showChildren}
+                         className='aggregation'>
+        <Typeahead placeholder='Enter state name or abbreviation'
+                   onInputChange={this._onInputChange}
+                   onOptionSelected={this._onOptionSelected}
+                   renderOption={this._renderOption} />
+        <ul>
+        {
+          this.props.options.map(bucket => 
+            <AggregationItem item={bucket}
+                             key={bucket.key}
+                             fieldName="state"
+            />
+          )
+        }
+        </ul>
+      </CollapsibleFilter>
+    )
+  }
+
+  // --------------------------------------------------------------------------
+  // Typeahead Helpers
+
+  _onInputChange(value) {
```

Comment:
Actually, the Typeahead component is perfectly happy to work on it's own, but each of these typeahead implementations are slightly different (compare these functions to Issue for an idea of the subtle difference).

Keep keeping me honest though!